src/main/java/com/ibm/cloud/sdk/core/security/TokenRequestBasedAuthenticator.java

```diff
@@ -211,6 +215,15 @@ public void authenticate(Builder builder) {
   public String getToken() {
     String token;
 
+    // Wait for token to get stored.
+    while (this.isTokenRequestInProgress) {
```

Comment:
Why don't you make `getToken()` synchronized?  See: https://docs.oracle.com/javase/tutorial/essential/concurrency/syncmeth.html

Reply:
That makes a lot of sense 🤦‍♀